### ISSUE TITLE
Digest mismatch, file must be downloaded again

### ISSUE THREAD

**Issue by tttt-0814** [ollama/ollama#3931]
### What is the issue?

I tried to pull nomic-embed-text, but got an error below.
I also tried to pull another models, but got the same error.

$ ollama pull nomic-embed-text
pulling manifest 
pulling 970aa74c0a90... 100% ▕███████████████████████████████████████████████████████████████████████████████████████████▏ 274 MB                         
pulling c71d239df917... 100% ▕███████████████████████████████████████████████████████████████████████████████████████████▏  11 KB                         
pulling ce4a164fc046... 100% ▕███████████████████████████████████████████████████████████████████████████████████████████▏   17 B                         
pulling 31df23ea7daa... 100% ▕███████████████████████████████████████████████████████████████████████████████████████████▏  420 B                         
verifying sha256 digest 
Error: digest mismatch, file must be downloaded again: want sha256:970aa74c0a90ef7482477cf803618e776e173c007bf957f635f1015bfcfef0e6, got sha256:bea7e365d4085c35b0cfc78f9285682c6c7df7a15ac58f1905542659871024fd

### OS

Linux

### GPU

Nvidia

### CPU

Intel

### Ollama version

0.1.32

**Comment by FlorianBoehler** [comment]
OS
Linux (Ubuntu 22.04.4 LTS)

Ollama version
0.1.32

Same issue:
Proxy is already set as Environment in ollama.service.

$ ollama pull nomic-embed-text
pulling manifest
pulling 970aa74c0a90... 100% ▕██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏ 274 MB
pulling c71d239df917... 100% ▕██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏  11 KB
pulling ce4a164fc046... 100% ▕██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏   17 B
pulling 31df23ea7daa... 100% ▕██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████▏  420 B
verifying sha256 digest
Error: digest mismatch, file must be downloaded again: want sha256:970aa74c0a90ef7482477cf803618e776e173c007bf957f635f1015bfcfef0e6, got sha256:bea7e365d4085c35b0cfc78f9285682c6c7df7a15ac58f1905542659871024fd

**Comment by FlorianBoehler** [comment]
A similar error occurs when trying to create a model from a .gguf file

$ ollama create Llama3:1 -f /home/user/downloads/Modelfile
transferring model data
Error: digest mismatch, expected "sha256:133b647485d69ae9458aba8448342b8c5495b92343f1df87fefbaf374ab4edc6", got "sha256:b86ae9d4b692d139592a7794ca507067d73e9e3ca2366fa7291c33a19c6254c6"

**Comment by TheQuantumbyte** [comment]
Same issue on MacOS trying to pull any model (phi3, llama3). It hangs at 100% pulling the model file for several minutes, then does the rest quickly and gives me the same digest mismatch error.

```
% ollama pull phi3              
pulling manifest 
pulling 4fed7364ee3e... 100% ▕███████████████████████████████████████▏ 2.3 GB                         
pulling c608dc615584... 100% ▕███████████████████████████████████████▏  149 B                         
pulling fa8235e5b48f... 100% ▕███████████████████████████████████████▏ 1.1 KB                         
pulling d47ab88b61ba... 100% ▕███████████████████████████████████████▏  140 B                         
pulling f7eda1da5a81... 100% ▕███████████████████████████████████████▏  485 B                         
verifying sha256 digest 
Error: digest mismatch, file must be downloaded again: want sha256:4fed7364ee3e0c7cb4fe0880148bfdfcd1b630981efa0802a6b62ee52e7da97e, got sha256:c9cc8bfb526a9ce885b566f7b034769ca850df757292424af8ad2406826324ec

```

**Comment by nb001** [comment]
I met the same situation.

**Comment by TheQuantumbyte** [comment]
Just as an added piece to the puzzle, I've been trying this on my work computer, which is running DNS filtering and other corporate software. I tried it at home with my personal machine and it worked correctly. I'm not sure if this failure is by design on the part of the DNS filtering solution or just an accident, but this was working fine last time I downloaded a model about 3 weeks ago.

**Comment by jmorganca** [comment]
merging this with https://github.com/ollama/ollama/issues/941

**Comment by ThomasGmeinder** [comment]
I still get this error. I just updated to the latest version of Ollama on Windows 11 and have been getting this error with the three previous versions too. The corporate network is not the issue because Ollama works fine for me on other Windows 11 machines. Command to reproduce: ollama run llama3.1:8b

**Comment by michaelrussell4** [comment]
I've had the same issue. This happens every time I try to download any model, the only exception and case where I've been able to download is `all-minilm`, maybe because it's only 45 MB?

Windows 11

**Comment by doug62** [comment]
This will fix it - for some reason:
echo "nameserver 8.8.8.8" > /etc/resolv.conf    # Thanks to -> https://www.youtube.com/watch?v=iSI-corX8HI

**Comment by keesj** [comment]
> This will fix it - for some reason: echo "nameserver 8.8.8.8" > /etc/resolv.conf # Thanks to -> https://www.youtube.com/watch?v=iSI-corX8HI

I do indeed see different behaviour. In my case I am using the snap package
